### PR TITLE
Customer billing is erased if the page is refreshed on checkout

### PR DESCRIPTION
**Pull request by kestraly**
I've changed this for billing.phtml in the persistent template to pull back data into the checkout from the quote for the billing address. billing.phtml within checkout/onepage template, however I'll only include this in the PR if other members feel this is an improved way of retrieving data back to the checkout and retaining customers.

### CODE REVIEW

File: app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml [hunk 1]
```diff
@@ -26,184 +26,184 @@
...
-        <?php $_streetValidationClass = $this->helper('customer/address')->getAttributeValidationClass('street'); ?>
-                <li class="wide">
-                    <label for="billing:street1" class="required"><em>*</em><?php echo $this->__('Address') ?></label>
-                    <div class="input-box">
-                        <input type="text" title="<?php echo Mage::helper('core')->quoteEscape($this->__('Street Address')) ?>" name="billing[street][]" id="billing:street1" value="<?php echo $this->escapeHtml($this->getAddress()->getStreet(1)) ?>" class="input-text <?php echo $_streetValidationClass ?>" />
-                    </div>
-                </li>
-        <?php $_streetValidationClass = trim(str_replace('required-entry', '', $_streetValidationClass)); ?>
-        <?php for ($_i = 2, $_n = $this->helper('customer/address')->getStreetLines(); $_i <= $_n; $_i++): ?>
+    <fieldset>
```
Comment (tmotyl) [inline app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml]: seems you have changed whitespace indentation. Is it necessary? it makes review much harder.
Can you please remove these chanegs and  update your PR ?

Comment (kestraly) [reply app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml]: Reviewing and reversing. Must of happened during a copy paste after moving it to the new branch

File: app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml [hunk 2]
```diff
@@ -45,14 +45,14 @@
                     <div class="field">
                         <label for="billing:company"><?php echo $this->__('Company') ?></label>
                         <div class="input-box">
-                            <input type="text" id="billing:company" name="billing[company]" value="<?php echo $this->escapeHtml($this->getAddress()->getCompany()) ?>" title="<?php echo Mage::helper('core')->quoteEscape($this->__('Company')) ?>" class="input-text <?php echo $this->helper('customer/address')->getAttributeValidationClass('company') ?>" />
+                            <input type="text" id="billing:company" name="billing[company]" value="<?php echo ($this->getAddress()->getCompany() ? $this->escapeHtml($this->getAddress()->getCompany()) : $this->getQuote()->getBillingAddress()->getCompany()); ?>" title="<?php echo Mage::helper('core')->quoteEscape($this->__('Company')) ?>" class="input-text <?php echo $this->helper('customer/address')->getAttributeValidationClass('company') ?>" />
```
Comment (kkrieger85) [inline app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml]: Is there a reason on why `$this->getQuote()->getBillingAddress()->getCompany()` is not echoed with `escapeHtml()`?

This also concerns all changes in this PR

**Comment by kestraly** [comment]
Mistake?

**Comment by kiatng** [comment]
Related to PR #969.

**Comment by joshua-bn** [comment]
What problem was this solving?

**Comment by kestraly** [comment]
@joshua-bn  - After proceeding past the billing step, if you refresh the checkout page, these details would be be reentered onto the checkout page.

**Comment by joshua-bn** [comment]
@kestraly they will be with this change or they are currently?

**Comment by kiatng** [comment]
@kestraly I think PR #969 fixed this. Please reopen if this is not the case.

**Pull request closed without merging**